### PR TITLE
size limit: max ID length

### PR DESCRIPTION
**Pull request by yiminc**
limit string length on various ID fields.
I did not make this dynamic configurable because 1K limit should be enough for all use cases. So this is more to protect some crazy long random string like 1MB of data rather than a control knob that we would want to tweak further.

### CODE REVIEW

File: common/util.go [hunk 1]
```diff
@@ -70,6 +70,10 @@ const (
 	FailureReasonDecisionBlobSizeExceedsLimit = "DECISION_BLOB_SIZE_EXCEEDS_LIMIT"
 	// TerminateReasonSizeExceedsLimit is reason to terminate workflow when history size exceed limit
 	TerminateReasonSizeExceedsLimit = "HISTORY_SIZE_EXCEEDS_LIMIT"
+
+	// MaxIDLengthLimit is the length limit for various IDs, including: Domain, TaskList, WorkflowID, ActivityID, TimerID,
+	// WorkflowType, ActivityType, SignalName, MarkerName, ErrorReason/FailureReason/CancelCause, Identity, RequestID
+	MaxIDLengthLimit = 1000 // max ID length limit
```
Comment (samarabbas) [inline common/util.go]: I feel this is similar to 'RPS' config knob and should be converted to a system level (no domain override) dynamic config.

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -701,13 +710,20 @@ func (wh *WorkflowHandler) PollForActivityTask(
...
 
+	if len(pollRequest.GetDomain()) > common.MaxIDLengthLimit {
+		return nil, wh.error(errDomainTooLong, scope)
+	}
+
 	if err := wh.validateTaskList(pollRequest.TaskList, scope); err != nil {
 		return nil, err
 	}
+	if len(pollRequest.GetIdentity()) > common.MaxIDLengthLimit {
+		return nil, wh.error(errIdentityTooLong, scope)
```
Comment (samarabbas) [inline service/frontend/workflowHandler.go]: Let's make sure we log enough information like DomainName, WorkflowID, RunID, etc when logging these errors.

File: common/util.go (hunk 1 above)
Comment (yiminc) [reply common/util.go]: changed to dynamic config

File: service/frontend/workflowHandler.go (hunk 2 above)
Comment (yiminc) [reply service/frontend/workflowHandler.go]: https://github.com/uber/cadence/issues/1364

**Pull request merged**